src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -414,6 +424,135 @@ private void LogMessage(string message, Exception e = null, string database = nu
+
+            if (currentResponsibleNode != null && newResponsibleNode.NodeTag == currentResponsibleNode)
+            {
+                if (notSuitableForTaskSince == null)
+                {
+                    // it's the same responsible node for backup, noop
+                    return null;
+                }
+
+                AddToDecisionLog(databaseName, $"Node '{currentResponsibleNode}' was in rehab for {DateTime.UtcNow - notSuitableForTaskSince}. " +
+                                               $"Since it's now in a member state, the backup will continue to run on that node");
+
+                // we need to remove the NotSuitableForTaskSince since the node is suitable for backup
+                return new ResponsibleNodeInfo
+                {
+                    TaskId = configuration.TaskId,
+                    ResponsibleNode = currentResponsibleNode,
+                    NotSuitableForTaskSince = null
+                };
+            }
+
+            switch (newResponsibleNode.Reason)
+            {
+                case ResponsibleNodeForBackup.ChosenNodeReason.MentorNode:
+                case ResponsibleNodeForBackup.ChosenNodeReason.PinnedMentorNode:
+                case ResponsibleNodeForBackup.ChosenNodeReason.NonExistingResponsibleNode:
+                case ResponsibleNodeForBackup.ChosenNodeReason.CurrentResponsibleNodeRemovedFromTopology:
+                    Debug.Assert(newResponsibleNode.ReasonForDecisionLog != null);
+                    AddToDecisionLog(databaseName, newResponsibleNode.ReasonForDecisionLog);
+                    return new ResponsibleNodeInfo
+                    {
+                        TaskId = configuration.TaskId,
+                        ResponsibleNode = newResponsibleNode.NodeTag
+                    };
+
+                case ResponsibleNodeForBackup.ChosenNodeReason.UnchangedResponsibleNode:
+                    // backward compatibility - we have a responsible node (the last which ran the backup)
+                    // but it's the first time that we save it in the storage
+                    return new ResponsibleNodeInfo
+                    {
+                        TaskId = configuration.TaskId,
+                        ResponsibleNode = newResponsibleNode.NodeTag
+                    };
+
+                case ResponsibleNodeForBackup.ChosenNodeReason.CurrentResponsibleNodeNotResponding:
+                    if (notSuitableForTaskSince == null)
+                    {
+                        // it's the first time that we identify that the node isn't suitable for backup
+                        AddToDecisionLog(databaseName, $"Node '{currentResponsibleNode}' is currently in rehab and cannot be used for backup. " +
+                                                       $"Will move the task to another node in {moveToNewResponsibleNodeGracePeriod}.");
```

Comment:
Add the actual time that this will happen?

Reply:
done